src/main.c

```diff
@@ -41,6 +43,8 @@ struct port_config {
 struct config {
 	float budget;
 	float budget_guard;
+	int i2c_bus;
```

Comment:
Which boards use I2C to communicate to the MCU for the [Broadcom-based PoE command frame implementation](https://svanheule.net/switches/software/broadcom_poe_control_protocol)? I ask because I wonder whether we can fetch the address and information at runtime from the devicetree.

Reply:
Added a column toe the [PoE management](https://svanheule.net/switches/software/poe_management) overview page to keep track of this info. I have a GS1920-24HP, but that's currently in use. To get the device info, realtek-poe could scan for certain compatibles. That means the MCU needs it's own node in the device tree, which is currently not the case.

Reply:
I would prefer that, too. One possibility to detect this, would also be to give the I2C bus in the .dts a nice name, such as "i2c-poe", these names show up in the /dev and /sys filesystems. It is further possible to probe that bus, in order to detect the right device address. Probing could be done within realtek-poe, there would not be the need to have additional tools outside.

Reply:
I'm going to look into .dts compatible bindings that we could use (or create!) to have the kernel begin buffering i2c reading for us. I doubt one exists for our specific case, but i2c is ubiquitous, so there's probably *something* out there.

Reply:
Similar to what @bkobl mentioned:

> No, I have searched the net and kernel for this. I2C calls are guaranteed to return in such a short amount of time it would anyway not make sense. The requirement to answer quickly is part of the I2C protocol in order not to block the I2C bus for other devices. This is different for a serial bus with 2 equal participants.

... there is no fancy binding out there to have the kernel call-back on code. There would be no point to having such a thing: a driver interacting with i2c is either entirely devoted to that i2c device, or can be written in a threaded manner so that blocking on an i2c read is no big deal.

Reply:
> Which boards use I2C to communicate to the MCU

@svanheule @bkobl: thanks for the [I2C column](https://svanheule.net/switches/software/poe_management), but I don't see any boards using I2C that have device trees in OpenWrt yet. I'm probably mistaken -- let me know about that so I can take a swing at adding this MCU to their device trees.

Reply:
No, I don't think you're mistaken. Most devices use the UART-variant, and neither of the two other devices from the table are in OpenWrt. My GS1920-24HP has been back at work since a while, as I sort of already had enough devices at home to play with (that have more useful bootloaders).

Reply:
The Netgear GS728TPv2 support is ready to be included but needs some other PRs to be merged first. It even has a working USB port! But frankly no one will use this device unless PoE works, so while PoE does not work I am not in a hurry. Personally, I need the GS728TPv2 at the end of the Summer to become the core switch at my Summer place driving plenty of sensors and WiFI repeaters over PoE. Until then it needs to work, but it also limits the time I can play around with it.